### ISSUE TITLE
How can I train a model without a GPU?

### ISSUE THREAD

**Issue by chy0428** [junyanz/pytorch-CycleGAN-and-pix2pix#1055]
Hello, this project is very interesting and awesome!

When I trained in colab, it took me over 7 hours to reach 50 epochs.
Is it right to take this long? Or is it a problem in my computer environment?

I tried to train this code on the Jupyter notebook. 
I do not have a GPU, so I entered `--gpu_ids -1` to use it as a CPU, and `visdom` was also installed and executed.
Afterwards, I ran the code of the training part, and the following logs were output, and my laptop was overheated and did not run.

```
----------------- Options ---------------
               batch_size: 1                             
                    beta1: 0.5                           
          checkpoints_dir: ./checkpoints                 
           continue_train: False                         
                crop_size: 256                           
                 dataroot: ./datasets/horse2zebra        	[default: None]
             dataset_mode: unaligned                     
                direction: AtoB                          
              display_env: main                          
             display_freq: 400                           
               display_id: 1                             
            display_ncols: 4                             
             display_port: 8097                          
           display_server: http://localhost              
          display_winsize: 256                           
                    epoch: latest                        
              epoch_count: 1                             
                 gan_mode: lsgan                         
                  gpu_ids: -1                            	[default: 0]
                init_gain: 0.02                          
                init_type: normal                        
                 input_nc: 3                             
                  isTrain: True                          	[default: None]
                 lambda_A: 10.0                          
                 lambda_B: 10.0                          
          lambda_identity: 0.5                           
                load_iter: 0                             	[default: 0]
                load_size: 286                           
                       lr: 0.0002                        
           lr_decay_iters: 50                            
                lr_policy: linear                        
         max_dataset_size: inf                           
                    model: cycle_gan                     
                 n_epochs: 100                           
           n_epochs_decay: 100                           
               n_layers_D: 3                             
                     name: horse2zebra                   	[default: experiment_name]
                      ndf: 64                            
                     netD: basic                         
                     netG: resnet_9blocks                
                      ngf: 64                            
               no_dropout: True                          
                  no_flip: False                         
                  no_html: False                         
                     norm: instance                      
              num_threads: 4                             
                output_nc: 3                             
                    phase: train                         
                pool_size: 50                            
               preprocess: resize_and_crop               
               print_freq: 100                           
             save_by_iter: False                         
          save_epoch_freq: 5                             
         save_latest_freq: 5000                          
           serial_batches: False                         
                   suffix:                               
         update_html_freq: 1000                          
                  verbose: False                         
----------------- End -------------------
dataset [UnalignedDataset] was created
The number of training images = 1334
initialize network with normal
initialize network with normal
initialize network with normal
initialize network with normal
model [CycleGANModel] was created
---------- Networks initialized -------------
[Network G_A] Total number of parameters : 11.378 M
[Network G_B] Total number of parameters : 11.378 M
[Network D_A] Total number of parameters : 2.765 M
[Network D_B] Total number of parameters : 2.765 M
-----------------------------------------------
Setting up a new session...
create web directory ./checkpoints/horse2zebra/web...
```

Why is training part not working, and how can I solve this problem?

**Comment by junyanz** [comment]
Which colab are you using? Which dataset? 7 hours on 50 epoches seems to be reasonable. 
CPU training is plausible but it will be extremely slow.

**Comment by chy0428** [comment]
Thank you for answer.
I ran the pytorch code on google colab and trained it using the horse2zebra dataset.
I tried training using another dataset.
The apple2orange dataset took 320 seconds in 1 epoch, which took about 16 hours in 200 epoch. However, in the case of colab, the session ends after 12 hours. Checkpoints were saved in the Jupyter notebook used locally, but the trained model isn't saved in the checkpoints in colab. I referenced this [QnA list](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/qa.md#can-i-continueresume-my-training-350-275-234-87 ) and other issues on this GitHub, but I couldn't get a solution.

![image](https://user-images.githubusercontent.com/47182864/83831098-a3145700-a721-11ea-87ff-b44f5130cf75.png)

![image](https://user-images.githubusercontent.com/47182864/83831137-b58e9080-a721-11ea-85f3-41a751ad5358.png)


How can I save checkpoints in colab?
Thank you.
